src/test/llmnode_test.cpp

```diff
@@ -132,19 +216,20 @@ TEST_F(LLMFlowHttpTest, unaryCompletionsJson) {
     ASSERT_EQ(
         handler->dispatchToProcessor(endpointCompletions, requestBody, &response, comp, responseComponents, &writer),
         ovms::StatusCode::OK);
-    rapidjson::Document d;
-    d.Parse(response.c_str());
-    ASSERT_TRUE(d["choices"].IsArray());
-    ASSERT_EQ(d["choices"].Capacity(), 1);
+    parsedResponse.Parse(response.c_str());
+    ASSERT_TRUE(parsedResponse["choices"].IsArray());
+    ASSERT_EQ(parsedResponse["choices"].Capacity(), 1);
     int i = 0;
-    for (auto& choice : d["choices"].GetArray()) {
+    for (auto& choice : parsedResponse["choices"].GetArray()) {
         ASSERT_TRUE(choice["finish_reason"].IsString());
-        ASSERT_EQ(choice["index"], i++);
+        EXPECT_STREQ(choice["finish_reason"].GetString(), "length");
```

Comment:
This is output dependent and might be an issue at some point. Let's avoid checking finish_reason content outside of specific finish_reason test.